pkg/ingester/pyroscope/ingest_adapter.go

```diff
@@ -152,6 +159,55 @@ func (p *pyroscopeIngesterAdapter) Evaluate(input *storage.PutInput) (storage.Sa
 	return nil, false // noop
 }
 
+func (p *pyroscopeIngesterAdapter) parseToPprof(ctx context.Context, in *ingestion.IngestInput, pprofable ingestion.ParseableToPprof) error {
+	plainReq, err := pprofable.ParseToPprof(ctx, in.Metadata)
+	if err != nil {
+		return fmt.Errorf("parsing IngestInput-pprof failed %w", err)
+	}
+	grpcReq, err := p.convertToGRPC(plainReq)
+	if err != nil {
+		return fmt.Errorf("converting IngestInput-pprof failed %w", err)
+	}
+	_, err = p.svc.Push(ctx, grpcReq)
+	if err != nil {
+		return fmt.Errorf("pushing IngestInput-pprof failed")
```

Comment:
looks like you're swallowing the error here ? Which mean this will always translate into 500 and should not because  there's more inside this error like validation and throttling.

Reply:
fixed